### DISCUSSION TITLE
How to using with Android Google Maps

### DISCUSSION THREAD

**Discussion by vTechGIS** [ngageoint/tiff-java#47] [Q&A] [answered]
I am looking for sample code to integrate this library into Google TileProvider.  Can you help me?  I want to read GeoTiff data and overlay it on Android Google Maps.

Thank you so much,

**Comment by bosborn** [comment, marked as answer]
[Here is an example](https://github.com/ngageoint/tiff-java/discussions/43#discussioncomment-1902423) of converting a RGB tiff to an Android Bitmap.
For Java, [here is an example](https://github.com/ngageoint/tiff-java/issues/4#issuecomment-283717483) of converting BufferedImage.

This library is currently only a basic TIFF specification implementation, pulling only those base parts from the Geotiff.js library. Other than [field tags](https://github.com/ngageoint/tiff-java/blob/master/src/main/java/mil/nga/tiff/FieldTagType.java#L235), GeoTIFF functionality is not currently implemented. You could try [following the origin and resolution implementations](https://github.com/geotiffjs/geotiff.js/blob/e31dfcc61dff03af5ac9209ad7a9b460c07acfc2/src/geotiffimage.js#L624) to pull that data out.

Some spec info on geotiff tags
http://geotiff.maptools.org/spec/geotiff2.6.html

We [use this library in Android](https://github.com/ngageoint/geopackage-android/search?q=import+mil.nga.tiff), but for a non visualization GeoPackage coverage data extension.  We have [non tiff image TileProviders](https://github.com/ngageoint/geopackage-android-map/tree/master/geopackage-map/src/main/java/mil/nga/geopackage/map/tiles/overlay) that might have some useful [tile bounds utilities](https://github.com/ngageoint/geopackage-core-java/blob/master/src/main/java/mil/nga/geopackage/tiles/TileBoundingBoxUtils.java#L324).

**Reply by vTechGIS** [reply]
Many thanks @bosborn
